test/lib/plaid_test.exs

```diff
@@ -154,6 +154,91 @@ defmodule PlaidTest do
     end
   end
 
+  describe "Telemetry events" do
+    @start_event [:plaid, :request, :start]
+    @stop_event [:plaid, :request, :stop]
+    @exception_event [:plaid, :request, :exception]
+
+    setup do
+      id = UUID.uuid4()
+      me = self()
+
+      :ok =
+        :telemetry.attach_many(
+          id,
+          [@start_event, @stop_event, @exception_event],
+          fn name, data, meta, :unused_config ->
+            send(me, {:telemetry_event, name, data, meta})
+          end,
+          :unused_config
+        )
+
+      on_exit(fn -> :telemetry.detach(id) end)
+    end
+
+    test "are sent on make_request/2", %{bypass: bypass} do
+      Bypass.expect(bypass, fn conn ->
+        Plug.Conn.resp(conn, 200, "{\"status\":\"ok\"}")
+      end)
+
+      {:ok, _resp} = Plaid.make_request(:get, "any")
+
+      [start, stop] = receive_events(2)
+
+      assert {@start_event, %{system_time: _}, start_meta} = start
+      assert {@stop_event, %{duration: _}, stop_meta} = stop
+
+      assert %{method: :get, path: "any", u: :native} = start_meta
+      assert %{method: :get, path: "any", status: 200, u: :native} = stop_meta
+    end
+
+    test "are sent when there's a lower level error", %{bypass: bypass} do
+      Bypass.down(bypass)
+
+      {:error, _econnrefused} = Plaid.make_request(:get, "any")
+
+      [start, stop] = receive_events(2)
+
+      assert {@start_event, %{system_time: _}, meta} = start
+      assert {@stop_event, %{duration: _}, ^meta} = stop
+
+      assert %{method: :get, path: "any", u: :native} = meta
+    end
+
+    test "are sent when there's an exception", %{bypass: bypass} do
+      # A character outside of utf-8 so that Poison raises
+      body = %{key: <<128>>}
```

Comment:
That's obviously not the best way to test that. The best would be to use Mox, have HTTP layer replacable but that would require to change all other tests. For know I think it's enough to use this little trick and when/if we introduce Mox we can change this place as well.